tools/__tasks__/validate/javascript.js

```diff
@@ -34,8 +30,26 @@ module.exports = {
         },
         {
             description: 'Run Flowtype checks on static/src/javascripts/',
-            task: `flow`,
-            onError: flowError,
+            task: ctx =>
+                execa('npm', ['run', 'flow'].concat(['--silent']))
+                    .then(res => {
+                        /* We end up here, if there is a problem with the flow
+                          server. The exit code would still 0, but flow logged
+                          something to res.stderr. In case flow found an error
+                          in the code it resolves within .catch() */
+                        if (res.stderr) {
+                            const err = new Error();
+                            err.stderr =
+                                'Apologies, flow server experienced a problem!';
+                            throw err;
+                        }
+                    })
+                    .catch(err => {
+                        ctx.messages
+                            .push(`Your editor may be able to catch flow errors as you work:\n${chalk.underline('https://docs.google.com/a/guardian.co.uk/document/d/1-w5KdwNVAZcGRL3Q9QCvj5y3aQyoVizm6GrHQaqQHNE/edit?usp=sharing')}`);
```

Comment:
update this link to go direct to the wiki page while you're here? https://github.com/guardian/frontend/wiki/So-you-want-to-ES6%3F